### ISSUE TITLE
reportUnknownMemberType not reported 

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#5839]
**Describe the bug**
reportUnknownMemberType is not reported in the code snippet below. Pyright behavior is a bit unstable, as removing seemingly unrelated lines makes the error appear.

**Code or Screenshots**

```python
# pyright: strict

from __future__ import annotations

from typing import Any


class Log:
    message: str
    extra: dict[str, Any]
    type: str


class Item:
    ...


def gather_data(items: dict[int, Item], logs: list[Log], changes: dict[str, dict[str, str]]):
    for log in logs:
        if log.type == "create":
            change = changes.setdefault("bar", {})
            change.setdefault("old", "bar")
        if log.type == "delete":
            old, _ = log.message.split(" -> ")
            change = changes.setdefault("bar", {})
        if log.type == "update":
            item = items[log.extra["id"]]
            old, _ = log.message.split(" -> ")
            change = changes.setdefault(f"{item.name}", {}).  # should report "Cannot access member "name" for type "Item"", but doesn't
            change.setdefault("old", old)
```

**VS Code extension or command-line**
pyright 1.1.324

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.325, which I just published. It will also be included in an upcoming pylance release.
